### DISCUSSION TITLE
`NumberInput` after `Combobox` have same ids

### DISCUSSION THREAD

**Discussion by daalfox** [chakra-ui/ark#2712] [General]
### Description

I put a `NumberInput` input after a `Combobox`. The `NumberInput` and `Combobox.Input` both are assigned the same id. It doesn't happen when I put the `NumberInput` **before** the `Combobox`. I have no idea why this is happening!

### Link to Reproduction (or Detailed Explanation)

I don't know how to reproduce this bug it's very weired

### Steps to Reproduce

I guess:
1. make a combobox
2. put a number input after it in the same div

### Ark UI Version

3.6.2

### Framework

- [ ] React
- [X] Solid
- [ ] Vue

### Browser

firefox 128.0.3

**Comment by segunadebayo** [comment]
Sounds weird. Could you share a reproduction @daalfox via Stackblitz

Perhaps it's a issue with Solid.js `createUniqueId`?

**Comment by daalfox** [comment]
> Sounds weird. Could you share a reproduction @daalfox via Stackblitz
> 
> Perhaps it's a issue with Solid.js `createUniqueId`?

Maybe. I try to get a reproduction working. See how it goes

**Reply by daalfox** [reply]
@segunadebayo I made some more investigations and it seems like having 2 inputs in a `Field.Input`'s `asChild` attribute causes them to have similar `id`s.
I decided to do it this way because in a project of mine there was a field for adding items to order and this items should specify 2 fields `variation` and `count`. It felt more natural this way because every item is essentially a field in form.
reproduction:
https://github.com/daalfox/ark-solid-reproduction/tree/main

I'm not sure if it's an issue with solid's `createUniqueId` or ark though.

**Reply by segunadebayo** [reply]
I see what you mean. It’s definitely not recommended to use multiple inputs within Field.Input. I recommend splitting them into separate fields or don’t use Field.Input at all.

The number input and combobox both read from the field context and end up having the same id. That’s why this is happening.

**Reply by daalfox** [reply]
Yeah I ended up using the later approach. Thanks for your help.
Btw your libraries are all awesome (panda, zagjs and ark). Keep up the good work!

**Comment by segunadebayo** [comment]
Closing — all components now generate unique ids per instance. This has been fixed.
